src/test/java/io/crate/consumer/CrateConsumer.java

```diff
@@ -0,0 +1,149 @@
+package io.crate.consumer;
+
+import com.carrotsearch.junitbenchmarks.AutocloseConsumer;
+import com.carrotsearch.junitbenchmarks.Result;
+import com.google.gson.JsonElement;
+import com.google.gson.JsonParser;
+import io.crate.testserver.action.sql.SQLRequest;
+import io.crate.testserver.action.sql.SQLResponse;
+import io.crate.testserver.client.CrateClient;
+import org.apache.log4j.Logger;
+
+import java.io.Closeable;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.InputStreamReader;
+import java.net.HttpURLConnection;
+import java.net.URL;
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.stream.Stream;
+
+public final class CrateConsumer extends AutocloseConsumer implements Closeable {
+
+    private final static Logger LOGGER = Logger.getLogger(CrateConsumer.class);
+
+    private final JsonParser jsonParser = new JsonParser();
+    private final DateFormat dateFormat = new SimpleDateFormat("yyyy-MM-dd'T'HH:mm:ssX");
+
+    private CrateClient client;
+
+    private String[] hosts;
+    private int transportPort;
+    private int httpPort;
+
+    public CrateConsumer() {
+        this.httpPort = getPort(CrateConsumerConstants.CRATE_HTTP_PORT);
+        this.transportPort = getPort(CrateConsumerConstants.CRATE_TRANSPORT_PORT);
+        this.hosts = getHosts();
+
+        client = new CrateClient(composeUrls(hosts, transportPort).toArray(String[]::new));
```

Comment:
are there no unit tests for the CrateConsumer? I'd make the CrateClient injectable so that you can mock it in tests